### PR TITLE
gmt4:  update to 4.5.17

### PR DESCRIPTION
**Pull request by fvaccari**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Addresses
[https://trac.macports.org/ticket/56433](url)

###### Tested on

macOS 10.12.6 16G1314
Xcode 9.2 9C40b 

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: science/gmt4/Portfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 PortSystem          1.0
 
 name                gmt4
-version             4.5.15
+version             4.5.17
 revision            1
```
Comment (mf2k) [inline science/gmt4/Portfile]: The `revision` line should be deleted because it starts at 0 when increasing the version and that is the default value.

Comment (mf2k) [reply science/gmt4/Portfile]: Also, please reference the Trac ticket in the commit message. Add this line to the end:

Closes: https://trac.macports.org/ticket/56433

**Comment by fvaccari** [comment]
I'm lost... I did all the stuff through the web interface, I don't see how to merge the two commits from there, and have no idea how to abandon the web and continue from a terminal. I think it might have been a better idea to use the terminal from the beginning. Sorry for all the mess...

**Comment by pmetzger** [comment]
Don't worry about it. I do think it's easier to use a terminal, because the web interface isn't very full featured, but we are in no rush.

**Comment by pmetzger** [comment]
@fvaccari Any progress on this? I'd like to integrate the change.

**Comment by fvaccari** [comment]
I’m sorry to say that having acted on the pull request through the web interface, and unfortunately quite in a chaotic way, I’ve no idea where in the universe those two commits that I should merge are hanging. Locally on my computer? Somewhere on a GitHub server?

Where do I have to go with Terminal to type the commands that you kindly suggested to me? 

> git reset --soft HEAD^^
> git commit -m "gmt4: update to 4.5.17"
> git push origin +master

**Comment by pmetzger** [comment]
@fvaccari Clearly nothing is locally on your computer unless you put it there. Have you ever used Git before? If not, you should read an introduction.

That said, I will do this update on your behalf later today.

**Comment by pmetzger** [comment]
The maintainer timed out, so I added this manually.

@fvaccari thanks for your contribution, but I'd suggest next time doing this almost entirely from the command line. It will be much easier.

**Comment by pmetzger** [comment]
Closed...

**Comment by pmetzger** [comment]
Closed.

**Comment by fvaccari** [comment]
I know of git what I needed to install bare git repositories on one of the servers I manage, with per-user permissions configured with gitolite. Many users from different clients with different permissions can clone the repositories, pull and push from there, create and merge branches etc, me included. 

On the other side, I’ve no experience at all with GitHub, and that is the root of the problem here…

I’m sorry to have created all of this mess, my intention was to give my infinitesimal contribution to MacPorts. Thanks for taking control from where I left...

As a side note, I think that such a basic operation like updating a port to a new minor release might deserve a dedicated and visible entry in the documentation, with an example sequence of commands that could be easily replicated by a newbie like myself. I started from Chapter 7.2 of the Guide and then jumped to Chapter 4. 

Steps listed in Section 4.5.1 seemed exactly what I was looking for,  but they do not mention the “revision" entry, that I left untouched, and was asked to remove after I committed my initial modification. This second commit was the reason I couldn’t manage to complete properly the pull request.

Maybe a small note could be added to 4.5.1 that, in case a simple action like moving to a new minor release of a package, the “revision” entry should be deleted from the Portfile.

Then, is there a way to do all the git stuff without going through the GitHub web interface? If so, an example session would be helpful, showing how to complete the task described in 4.5.1.

I hope I’ll do a better job when gmt 4.5.18 will be released… :-/

**Comment by fvaccari** [comment]
Definitely, that is what I was asking in the comment I wrote while you were typing this… :-)

**Comment by mojca** [comment]
The commit which replaced this PR: d5949028bb30dcf49e3fd888de441a7fb6a194c3

**Comment by pmetzger** [comment]
@fvaccari if you could contribute some updated language for the guide, that would be terrific. We're always looking to make the manual better, and helping us make things clearer when your confusion is fresh in your mind is always the best time to do it.

**Comment by fvaccari** [comment]
@pmetzger Well, my confusion is so fresh, that I still don’t have a clue of how I should have merged the two commits myself. Go figure how good a manual I could write…. :-)

In my experience, it’s really worth to spend some time on writing **concise** documentation describing very specific tasks. It will pay back in terms of time not spent again and again in explaining the same thing to different users, who may want to try to contribute, but get scared by the complexity. And I believe it’s only the expert who can write good docs. 

MacPorts is huge, so being concise is not really an easy task. Along the same line, I find myself dealing with many powerful and complicated software, of which I use just the small percent of capabilities that allows me to get the job done. Here I feel the need to identify that specific small percent of git that I may need to learn deeply, when I decide to be brave and try to contribute to MacPorts. 

And I need examples of sequences of actions needed to complete the most common tasks, like a point-update of a package definitely is. Being pointed to a 500+ pages book on git is hardly what can help me getting the job done quickly. 

Of course, don't take this as a criticism, as I do really appreciate your patience and the help I've got. I'm trying just to focus on what an unexperienced user like me is looking for in the docs...

**Pull request closed without merging**